src/pgduckdb_planner.cpp

```diff
@@ -32,11 +32,11 @@ extern "C" {
 bool duckdb_explain_analyze = false;
 
 duckdb::unique_ptr<duckdb::PreparedStatement>
-DuckdbPrepare(const Query *query) {
+DuckdbPrepare(const Query *query, bool allow_explain) {
 	Query *copied_query = (Query *)copyObjectImpl(query);
 	const char *query_string = pgduckdb_get_querydef(copied_query);
 
-	if (ActivePortal && ActivePortal->commandTag == CMDTAG_EXPLAIN) {
+	if (allow_explain && ActivePortal && ActivePortal->commandTag == CMDTAG_EXPLAIN) {
```

Comment:
Should't the condition be:
```suggestion
	if (ActivePortal && ActivePortal->commandTag == CMDTAG_EXPLAIN) {
	  if (!allow_explain) {
	    elog(ERROR, ...);
	  }
```

Reply:
No it shouldn't, but I totally agree that the argument name suggested that. Instead of renaming it I moved the logic outside of `DuckdbPrepare` function now.